.github/workflows/cmake.yml

```diff
@@ -0,0 +1,45 @@
+name: CMake
+
+on:
+  push:
+    branches:
+      - master
+    paths-ignore:
+      - '**.md'
+  pull_request:
+    branches:
+      - master
+    paths-ignore:
+      - '**.md'
+
+env:
+  BUILD_TYPE: Debug
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+
+    runs-on: ${{matrix.os}}
+
+    steps:
+    - uses: actions/checkout@v2
```

Comment:
not quite understanding the syntax here even with the help of the [docs](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#example-using-versioned-actions).

`actions/checkout@v2`  I think this is checking out the branch of the repo from the push/pull request, but why version `2`?

Reply:
I just cut & pasted from the GH Actions docs when I started playing with it, but I assume @v2 is the current version of the builtin action and somehow incompatible with the previous version.

Reply:
Indeed, `uses` clauses can use complex (or simple) actions provided by others. Instead of manually providing `git clone` as a shell command, one can import that in a slightly more robust way with customization points exposed as parameters to `uses`. Introduction [here](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/introduction-to-github-actions#understanding-the-workflow-file) and the specific action and it's parameters [here](https://github.com/marketplace/actions/checkout)

Reply:
I see, thanks for the links @MathiasMagnus :+1: